### ISSUE TITLE
Lightmap Directional Disables Bicubic Interpolation

### ISSUE THREAD

**Issue by WickedInsignia** [godotengine/godot#88470]
### Tested versions

4.3 Dev3

### System information

Windows 11, Nvidia RTX4070ti, AMD Ryzen7700x

### Issue description

Activating Directional in LightmapGI disables Bicubic Interpolation.

Directional on:
![GodotDirectionalOnBicubic](https://github.com/godotengine/godot/assets/70243134/7f2b9268-6095-4308-bf94-e634e18a348c)

Directional off:
![GodotDirectionalOffBicubic](https://github.com/godotengine/godot/assets/70243134/c2780abe-0080-4edf-bdb7-abf58f859d7d)

### Steps to reproduce

Turn on Directional in Lightmap GI.

### Minimal reproduction project (MRP)

Not needed. Just turn the setting on in any scene.

**Comment by Calinou** [comment]
There's no bicubic interpolation to disable because [it hasn't been reimplemented in 4.x yet](https://github.com/godotengine/godot/issues/49935) :slightly_smiling_face: 

What you're pointing out is likely the DirectionalLight **Angular Distance** property not being taken into account when baking directional lightmaps.

**Comment by Calinou** [comment]
I can confirm this on 4.3.beta 92c8e87cd. This is not a tonemapping issue: it stays regardless of which tonemapper is used, and is present in all rendering methods. It also occurs even if the light's **Angular Distance** is `0.0`.

Directional disabled | Directional enabled
-|-

![image](https://github.com/godotengine/godot/assets/180032/884130aa-c9f7-4399-9a75-0bdccb4279c7) | ![image](https://github.com/godotengine/godot/assets/180032/72bff5b3-f07b-4af6-ad88-94e0d6d2e21d)

**Comment by clayjohn** [comment]
I just took a look at this, it appears that the issue is the denoiser isn't smoothing out the aliasing as much when using the directional mode. 

They look the same with denoising turned off. This should be an easy fix.

Additionally, it looks way better with https://github.com/godotengine/godot/pull/95828 and https://github.com/godotengine/godot/pull/89919

**Comment by clayjohn** [comment]
Turns out it is not an easy fix. I still can't find why its not working. However, I can confirm that the problem is only with the JNLM denoiser, using OIDN works fine

**Comment by Calinou** [comment]
https://github.com/godotengine/godot/pull/89919 doesn't seem to work on directional lightmaps from my testing.

**Comment by Calinou** [comment]
Directional lightmaps seem to work with bicubic sampling in `master` and with https://github.com/godotengine/godot/pull/95828: https://github.com/godotengine/godot/pull/95828#pullrequestreview-2249231440

**Comment by clayjohn** [comment]
I think I've figured out the root cause of this issue, it has to do with how we scale the spherical harmonics. We are baking the directional lightmaps way too bright and then scaling back at run time. I'm trying to figure out the proper scaling factors now. 

This will impact the work that @BlueCube3310 is doing on scaling the L1 spherical harmonics for compression

**Comment by Calinou** [comment]
> We are baking the directional lightmaps way too bright and then scaling back at run time.

Yes, I remember discussing this in https://github.com/godotengine/godot/pull/61910. It turns out what I was doing was actually wrong (which was our initial guess).

**Comment by clayjohn** [comment]
Using the coefficients from https://media.contentapi.ea.com/content/dam/eacom/frostbite/files/gdc2018-precomputedgiobalilluminationinfrostbite.pdf we can get things to look the same between directional and non-directional!

|Before directional | Before non-directional | After directional | After directional (no normal map) |
| - | - | - | - |
| ![Screenshot from 2024-08-20 16-41-12](https://github.com/user-attachments/assets/fbfc7605-762e-47f6-b22a-df03dc608f55) | ![Screenshot from 2024-08-20 16-37-18](https://github.com/user-attachments/assets/d4720d16-3a0d-4067-8031-e0b12f2bfd70) | ![Screenshot from 2024-08-20 16-40-20](https://github.com/user-attachments/assets/4aa40c5d-49a1-4717-9e60-077455499a2a) | ![image](https://github.com/user-attachments/assets/de57437b-183a-429c-8b3a-a823409532ad) |
